### PR TITLE
Split and improve recently created test module install_patterns

### PR DESCRIPTION
**Pull request by manfredi**
- Description: Split and improve recently created test module install_patterns
- Related ticket: [issues/132911](https://progress.opensuse.org/issues/132911)
- Verification run: [Test result overview](https://openqa.suse.de/tests/overview?version=15-SP5&distri=sle&build=manfredi%2Fos-autoinst-distri-opensuse%23issues-132911)
- Related [MR#319](https://gitlab.suse.de/coolgw/wegao-test/-/merge_requests/319)

### CODE REVIEW

**Review by rakoenig — APPROVED** [review]
LGTM.

File: schedule/migration/migration_offline_media_x86_64_all.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: migration_offline_x86_64.yaml
```
Comment (lemon-suse) [inline schedule/migration/migration_offline_media_x86_64_all.yaml:1]: It's better to update the name for 'all patterns',
Ex: schedule/migration/migration_offline_media_all_patterns_x86_64.yaml

File: schedule/migration/migration_offline_scc_x86_64_all.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: migration_offline_x86_64.yaml
```
Comment (lemon-suse) [inline schedule/migration/migration_offline_scc_x86_64_all.yaml:1]: Suggest to update name:
schedule/migration/migration_offline_scc_all_patterns_x86_64.yaml

File: tests/migration/install_all_patterns.pm [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Install patterns for allpatterns cases before conducting migration
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
```
Comment (lemon-suse) [inline tests/migration/install_all_patterns.pm:12]: Please just include functions used here.

File: tests/migration/install_all_patterns.pm [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+use utils;
```
Comment (lemon-suse) [inline tests/migration/install_all_patterns.pm:13]: It seems we don't need to include it.

File: tests/migration/query_all_rpm.pm [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Record the installed rpm list before conducting migration
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
```
Comment (lemon-suse) [inline tests/migration/query_all_rpm.pm:12]: Please just include functions used here.

**Comment by lemon-suse** [comment]
Please run a test for default patterns( do not include your new modules in schedule file ) to avoid regression, ex: https://openqa.suse.de/tests/11605772#

File: tests/migration/install_all_patterns.pm (hunk 3 above)
Comment (manfredi) [reply tests/migration/install_all_patterns.pm:12]: Could you explain me better what we need to include? At the moment it is included only the used modules

Comment (lemon-suse) [reply tests/migration/install_all_patterns.pm:12]: In your test module, only used select_console, so include the function of select_console is enough.

File: schedule/migration/migration_offline_media_x86_64_all.yaml (hunk 1 above)
Comment (manfredi) [reply schedule/migration/migration_offline_media_x86_64_all.yaml]: rename to `schedule/migration/migration_offline_media_all_patterns_x86_64.yaml`

File: schedule/migration/migration_offline_scc_x86_64_all.yaml (hunk 2 above)
Comment (manfredi) [reply schedule/migration/migration_offline_scc_x86_64_all.yaml]: rename to `schedule/migration/migration_offline_scc_all_patterns_x86_64.yaml`

**Comment by manfredi** [comment]
> Please run a test for default patterns( do not include your new modules in schedule file ) to avoid regression, ex: https://openqa.suse.de/tests/11605772#

Result: passed: https://openqa.suse.de/tests/11704426#

File: schedule/migration/migration_offline_media_all_patterns_x86_64.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+name: migration_offline_x86_64.yaml
```
Comment (lemon-suse) [inline schedule/migration/migration_offline_media_all_patterns_x86_64.yaml:1]: This line need to be updated to new name also.

File: schedule/migration/migration_offline_scc_all_patterns_x86_64.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+name: migration_offline_x86_64.yaml
```
Comment (lemon-suse) [inline schedule/migration/migration_offline_scc_all_patterns_x86_64.yaml:1]: This line need to be updated to new name also.

Comment (manfredi) [reply schedule/migration/migration_offline_scc_all_patterns_x86_64.yaml]: Updated to `name: migration_offline_scc_all_patterns_x86_64.yaml`

File: schedule/migration/migration_offline_media_all_patterns_x86_64.yaml (hunk 6 above)
Comment (manfredi) [reply schedule/migration/migration_offline_media_all_patterns_x86_64.yaml]: Updated to `name: migration_offline_media_all_patterns_x86_64.yaml`

**Comment by lemon-suse** [comment]
> > Please run a test for default patterns( do not include your new modules in schedule file ) to avoid regression, ex: https://openqa.suse.de/tests/11605772#
> 
> Result: passed: https://openqa.suse.de/tests/11704426#

You should run the test with your PR and remove the install_patterns from the yaml schedule, so that will be a test with default patterns. Note, after split the modules, with the module in yaml schedule means all patterns; without that module means default patterns.

**Review by lemon-suse — APPROVED** [review]

**Pull request merged**